test/movies-routes-test.js

```diff
@@ -0,0 +1,245 @@
+'use strict';
+
+const expect = require('chai').expect;
+const request = require('superagent');
+const Movies = require('../model/movies.js');
+const PORT = process.env.PORT || 3000;
+
+require('../server.js');
+
+const url = `http://localhost:${PORT}`;
+
+const exampleMovie = {
+  name: 'Example Movie',
+  rating: 0
+};
+
+describe('Movie Routes', function() {
+  describe('POST: /api/movies', function() {
+    describe('Valid Body', function() {
+      after( done => {
+        if(this.tempMovie) {
+          Movies.findByIdAndRemove(this.tempMovie._id)
+            .then( () => done())
+            .catch(done);
+          return;
+        }
+        done();
+      });
+      it('should respond with a status of 200', done => {
+        request.post(`${url}/api/movies`)
+          .send(exampleMovie)
+          .end((err, res) => {
+            if(err) return done(err);
+            expect(res.status).to.equal(200);
+            this.tempMovie = res.body;
+            done();
+          });
+      });
+      it('should have properties of name, rating, timestamp, id, and directorID', done => {
+        expect(this.tempMovie).to.include.keys('name');
```

Comment:
Tests are awesome! Way to write 48 of them- totally sweet. There's a shorter way to check for all the keys, although the way you are doing it works totally fine, you could have also done something like:

expect({ foo: 1, bar: 2 }).to.have.all.keys(['bar', 'foo']);